pkg/scheduler/core/generic_scheduler.go

```diff
@@ -697,6 +698,9 @@ func PrioritizeNodes(
 		errs []error
 	)
 	appendError := func(err error) {
+		if !alwaysCheckAllPredicates && len(errs) > 0 {
```

Comment:
This doesn't look right. In priority phase, all registered priorities must be went through on filtered nodes which have passed predicates phase.

Reply:
This check only affects the appending of errors.
It wouldn't skip any priority.

Reply:
It doesn't look right either - atm `appendError` unconditionally append the error and gets aggregated and return. Why (1) it has to do with `alwaysCheckAllPredicates`? and (2) you want to change the behavior to only return one error?

Reply:
We can introduce another flag, if needed.

For #1, @misterikkit mentioned this flag in #78208.

Reply:
I guess Jonathan somehow misread it. `alwaysCheckAllPredicates` is only related with Predicates. And in 99.99% cases, it's left as default value (false).